### PR TITLE
Common empty device list

### PR DESCRIPTION
**Pull request by atrovato**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- ~~[ ] If your changes affects code, did your write the tests?~~
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- ~~[ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)~~
- ~~[ ] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.~~
- ~~[ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~~
- ~~[ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~~
- ~~[ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).~~

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Unify empty device list component accross integration services.

---
@Pierre-Gilles I think we should improve the developer documentation for common components.

### CODE REVIEW

File: front/src/routes/integration/all/philips-hue/device-page/FoundDevices.jsx [hunk 1]
```diff
@@ -74,8 +73,10 @@ const FoundDevices = ({ children, ...props }) => (
                   </div>
                 </div>
               ))}
-            {props.philipsHueDevices && props.philipsHueDevices.length === 0 && (
-              <Text id="integration.philipsHue.device.noDevices" />
+            {(!props.philipsHueDevices || props.philipsHueDevices.length === 0) && (
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/philips-hue/device-page/FoundDevices.jsx:78]: Philips hue empty state is added 2 times ? 

Screenshot 2022-04-01 at 15 36 15

File: front/src/routes/integration/all/xiaomi/DevicePanel.jsx [hunk 2]
```diff
@@ -57,10 +58,10 @@ const DevicePanel = ({ children, ...props }) => (
                   deleteDevice={props.deleteDevice}
                 />
               ))}
-            {props.xiaomiDevices && props.xiaomiDevices.length === 0 && (
-              <p class="text-center">
+            {(!props.xiaomiDevices || props.xiaomiDevices.length === 0) && (
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/xiaomi/DevicePanel.jsx:61]: Why did you change the condition ? 
Not sure this is the goal of the empty state, empty state != loading state

Comment (Pierre-Gilles) [reply front/src/routes/integration/all/xiaomi/DevicePanel.jsx:61]: The empty state should be displayed only when the state is indeed empty, not loading

File: front/src/routes/integration/all/zigbee2mqtt/device-page/DeviceTab.jsx [hunk 3]
```diff
@@ -52,7 +52,11 @@ const DeviceTab = ({ children, ...props }) => (
               props.zigbee2mqttDevices.map((device, index) => (
                 <Zigbee2mqttBox {...props} device={device} deviceIndex={index} />
               ))}
-            {props.zigbee2mqttDevices && props.zigbee2mqttDevices.length === 0 && <EmptyState />}
+            {(!props.zigbee2mqttDevices || props.zigbee2mqttDevices.length === 0) && (
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/zigbee2mqtt/device-page/DeviceTab.jsx:54]: Same here.

File: front/src/routes/integration/all/zigbee2mqtt/device-page/DeviceTab.jsx [hunk 4]
```diff
@@ -52,7 +52,11 @@ const DeviceTab = ({ children, ...props }) => (
...
+              <EmptyDeviceList>
+                <Text id="integration.zigbee2mqtt.noDeviceFound" />
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/zigbee2mqtt/device-page/DeviceTab.jsx:56]: The empty state was better before

Now: 
Screenshot 2022-04-01 at 15 40 26

Before:
Screenshot 2022-04-01 at 15 40 22

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]

File: front/src/routes/integration/all/xiaomi/DevicePanel.jsx (hunk 2 above)
Comment (atrovato) [reply front/src/routes/integration/all/xiaomi/DevicePanel.jsx:61]: The loading state should display the loader div inside the dimmer block.
This part is should be displayed only once the getXiaomiDevicesStatus !== Getting.

**Comment by atrovato** [comment]
I will take it back and split it for each integration pages with screenshots, to make it easier to review and test.
Keep tracking on https://github.com/atrovato/Gladys/pull/28
